components/inputs/test/input-number.test.js

```diff
@@ -408,6 +408,7 @@ describe('d2l-input-number', () => {
 				const errors = await elem.validate();
 				if (test.expectedError) expect(errors).to.contain(test.expectedError);
 				else expect(errors).to.be.empty;
+				expect(elem.invalid).to.equal(!!test.expectedError);
```

Comment:
There's already a test for the `required` case - we just need to check the assert the `invalid` state. Good to check this for all the cases. 😄